lib/client/conntest/ssh.go

```diff
@@ -49,30 +59,239 @@ func NewSSHConnectionTester(cd services.ConnectionsDiagnostic) *SSHConnectionTes
 //   - the SSH Node will append a trace indicating if the has access (RBAC)
 //   - the SSH Node will append a trace indicating if the requested principal is valid for the target Node
 func (s *SSHConnectionTester) TestConnection(ctx context.Context, req TestConnectionRequest) (types.ConnectionDiagnostic, error) {
-	id := uuid.NewString()
-	connectionDiagnostic, err := types.NewConnectionDiagnosticV1(id, map[string]string{},
+	if req.ResourceKind != types.KindNode {
+		return nil, trace.BadParameter("invalid value for ResourceKind, expected %q got %q", types.KindNode, req.ResourceKind)
+	}
+
+	connectionDiagnosticID := uuid.NewString()
+	connectionDiagnostic, err := types.NewConnectionDiagnosticV1(connectionDiagnosticID, map[string]string{},
 		types.ConnectionDiagnosticSpecV1{
-			Message: types.DiagnosticMessageWaiting,
+			Message: types.DiagnosticMessageFailed,
 		})
+	if err != nil {
+		return nil, trace.Wrap(err, "new conn diag")
+	}
+
+	if err := s.clt.CreateConnectionDiagnostic(ctx, connectionDiagnostic); err != nil {
+		return nil, trace.Wrap(err, "create conn diag")
```

Comment:
If we return in any of these error cases, a trace won't be created. In case of any error, can we create a failure trace with an appropriate error message?

Reply:
The only error I see as being helpful here is when we don't have permissions for the `connection_diagnostic` resource :thinking: 

I can add a check here and before proceeding, it will create, update and read the Connection Diagnostic.
If any of those operations fails with a `trace.IsAccessDenied`, we return a trace with that info.

Everything else should be treated as "normal" server error.

What do you think?